### ISSUE TITLE
Set permissions still inherited from collection permissions

### ISSUE THREAD

**Issue by atomrab** [benwbrum/fromthepage#1192]
This may be a UT-specific problem related to working with students, but I'd like to know if the behavior is a) expected and b) on deck to be changed. 

My use-case: I want to allow students to transcribe a specific private set of documents without giving them access to the whole collection in which those documents are located, because I'm trying to prevent them from accidentally transcribing other documents (this usually happens, even if I point them at a set). 

The behavior: although the description of the "add collaborator" function in a private set suggests that adding a collaborator gives access to that set, collaborators who have been added are not able to transcribe until they're also added to the collection as a whole. So the privacy setting works to keep a set hidden from the public, but it doesn't constrain access for collaborators, since the only people who can transcribe its documents must also have access to all the other documents in the collection.

This is only a minor issue, but I want to make sure that private sets are supposed to inherit permissions from the main collection, and I'm curious about plans to make it possible to add collaborators *only* to a given private set without adding them to the main collection. This would be useful in guiding work with relatively inexperienced collaborators.

**Comment by atomrab** [comment]
Actually, this seems to be more problematic than I thought. The interface now has several different options for authorization: one can be a collection owner, a collection collaborator, and a private set collaborator. Initially I thought that I could just add students as private set collaborators, so that they could work on only the documents I chose for them. Then it became apparent that they had to be added as collaborators to the main collection before they could see the private sets, so I added them there -- as collaborators, not owners, because I thought that this category had been added to avoid giving transcribers control over the collection as a whole.

It appears, however, that the only way I can allow students access to transcription in a private collection or private sets from a private collection is still to make them collection owners. Is this because having different permissions was too difficult, or is this an unresolved bug that there are plans to fix for the UT version or the general version or both?

**Comment by atomrab** [comment]
Addendum: one reason this permissions situation confuses students is that they like to go straight to the dashboard, not the Collections page, assuming that the dashboard will tell them what to do. If they're added as collaborators, their dashboard is entirely blank -- it doesn't show collections or sets they've been added to as collaborators, though if they could transcribe something, it might show work they've already done. 

If, however, they're added as owners, then they just see a list of all the works in the collection, in order. I suspect this is why they often end up transcribing random works, rather than works in the sets I make for them. 

Fixing permissions might help this, but it might also be worth considering a minor UI adjustment: a collaborator dashboard should show all collections or sets a collaborator is able to transcribe, rather than only what they've already worked on. If you have a lot of novice users who expect the dashboard to tell them what to do or where to go, this simple change could help.

**Comment by saracarl** [comment]
Hi Adam.  I just tried this and it is working exactly as we both expect.  

Here's the private document set I created with Ben's account (which was also an owner) on your private test collection:  https://fromthepage.lib.utexas.edu/benwbrum/sara-test-of-access 
I then added "saracarl" as a collaborator, logged out, logged in as saracarl, and was able to transcribe.
https://fromthepage.lib.utexas.edu/benwbrum/sara-test-of-access

Can we have a call to figure out what's going on?  Either it's a remnant in the database (your collections, and possibly document sets, were created before we wrote most of this functionality) or it's just confusing to set up.  Either way, I think we can get this working.
